source/js/plugins/simple-auth.js

```diff
@@ -0,0 +1,206 @@
+import { elt } from '../utils/elt';
+
+/**
+ * A simple authentication plugin. 
+ * 
+ * Based on parts of IIIF-Auth API but with access control on the manifest. 
+ * 
+ * Requires IIIF-Auth API compliant Access Cookie Service (settings.simpleAuthLoginUrl)
+ * and Access Token Service (settings.simpleAuthTokenUrl).
+ * (see https://iiif.io/api/auth/1.0/)
+ * 
+ * 1. Diva tries to load the manifest normally.
+ * 2. If loading the manifest fails with an authorization error the plugin requests an access token.
+ * 3. If loading the access token fails the plugin shows a login dialog and requests a new token.
+ * 4. Diva tries to load the manifest with the access token.
+ * 5. If loading the manifest fails with the access token goto 3.
+ * 6. The images are loaded using the cookies set by the login domain.
+ *  
+ **/
+export default class SimpleAuthPlugin
+{
+    constructor (core)
+    {
+        this.core = core;
+        this.authTokenUrl = core.settings.simpleAuthTokenUrl;
+        this.authLoginUrl = core.settings.simpleAuthLoginUrl;
+        this.authToken = null;
+        this.authError = null;
+        this.authTokenId = null;
+        this.serviceOrigin = this.getOrigin(this.authTokenUrl);
+
+        /*
+         * Manifest load error handler.
+         * 
+         * Re-tries load with auth token or shows log in window. 
+         */
+        Diva.Events.subscribe("ManifestFetchError", (response) => {
+        	if (response.status == 401) {
+            	if (this.authToken == null && this.authError == null) {
+            		// no auth token. let's get one
+            		this.requestAuthToken();
+            		// abort regular error message
+            		throw new Error("cancel error handler");
```

Comment:
This should have a more descriptive error message

Reply:
I'll try. The problem is that I throw this exception as a way to abort the existing error message in `_AjaxError()` so it is not really an error at this point :-)

The cleanest solution would be to have the error handling be part of the fetch Promise chain. Do you have an idea how to do this? Can I post a Promise with the `ManifestFetchError` event and resolve it directly?